demos/python_demos/face_recognition_demo/face_recognition_demo.py

```diff
@@ -0,0 +1,1056 @@
+
+        def __getitem__(self, idx):
+            return self.points[idx]
+
+    def __init__(self, model):
+        super(LandmarksDetector, self).__init__(model)
+
+        assert len(model.inputs) == 1, "Expected 1 input blob"
+        assert len(model.outputs) == 1, "Expected 1 output blob"
+        self.input_blob = next(iter(model.inputs))
+        self.output_blob = next(iter(model.outputs))
+        self.input_shape = model.inputs[self.input_blob].shape
+
+        assert np.array_equal([1, self.POINTS_NUMBER * 2, 1, 1],
+                model.outputs[self.output_blob].shape), \
+            "Expected model output shape %s, but got %s" % \
+            ([1, self.POINTS_NUMBER * 2, 1, 1],
+             model.outputs[self.output_blob].shape)
+
+    def preprocess(self, frame, rois):
+        assert len(frame.shape) == 4, "Frame shape should be [1, c, h, w]"
+        inputs = cut_rois(frame, rois)
+        inputs = [self._resize(input, self.input_shape) for input in inputs]
+        return inputs
+
+    def enqueue(self, input):
+        return super(LandmarksDetector, self).enqueue({self.input_blob: input})
+
+    def start_async(self, frame, rois):
+        inputs = self.preprocess(frame, rois)
+        for input in inputs:
+            self.enqueue(input)
+
+    def get_landmarks(self):
+        outputs = self.get_outputs()
+        results = [ self.Result(out[self.output_blob].reshape((-1, 2))) \
+                    for out in outputs ]
+        return results
+
+
+class FacesDatabase:
+    IMAGE_EXTENSIONS = ['.jpg', '.png']
+
+    class Identity:
+        def __init__(self, label, descriptors):
+            self.label = label
+            self.descriptors = descriptors
+
+    def __init__(self, path,
+            face_identifier, landmarks_detector, face_detector=None):
+        path = osp.abspath(path)
+        self.fg_path = path
+        paths = []
+        if osp.isdir(path):
+            ext = self.IMAGE_EXTENSIONS
+            paths = [osp.join(path, f) for f in os.listdir(path) \
+                      if f.endswith(ext[0]) or f.endswith(ext[1])]
+        else:
+            raise Exception("Wrong face images database path. Expected a " \
+                            "path to the directory containing %s files, " \
+                            "but got '%s'" % \
+                            (" or ".join(self.IMAGE_EXTENSIONS), path))
+
+        if len(paths) == 0:
+            raise Exception("The images database folder has no images")
+
+        self.database = []
+        for num, path in enumerate(paths):
+            label = osp.splitext(osp.basename(path))[0]
+            image = cv2.imread(path, flags=cv2.IMREAD_COLOR)
+
+            assert len(image.shape) == 3, \
+                "Expected an input image in (H, W, C) format"
+            assert image.shape[2] in [3, 4], \
+                "Expected BGR or BGRA input"
+
+            orig_image = image.copy()
+            if image.shape[0] == 4: # assume BGRA
```

Comment:
Wrong shape index here. We can either use 2 (as above) or, probably, just remove the line.